File: frontend/lib/features/transport/data/models/vehicle.dart
import 'transport_status.dart';

class Vehicle {
  const Vehicle({
    required this.id,
    required this.schoolId,
    required this.schoolName,
    required this.name,
    required this.registrationNumber,
    required this.capacity,
    required this.status,
    required this.routeId,
    required this.routeName,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) {
    return Vehicle(
      id: json['id'] as int,
      schoolId: json['school_id'] as int,
      schoolName: json['school_name'] as String?,
      name: json['name'] as String,
      registrationNumber: json['registration_number'] as String,
      capacity: json['capacity'] as int,
      status: TransportStatus.fromApiValue(json['status'] as String),
      routeId: json['route_id'] as int?,
      routeName: json['route_name'] as String?,
    );
  }

  final int id;
  final int schoolId;
  final String? schoolName;
  final String name;
  final String registrationNumber;
  final int capacity;
  final TransportStatus status;

  /// The route this vehicle currently serves, if any.
  final int? routeId;
  final String? routeName;

  bool get isActive => status == TransportStatus.active;
}

File: backend/app/Models/Vehicle.php
<?php

namespace App\Models;

use App\Enums\TransportStatus;
use Database\Factories\VehicleFactory;
use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasOne;

#[Fillable(['school_id', 'name', 'registration_number', 'capacity', 'status'])]
class Vehicle extends Model
{
    /** @use HasFactory<VehicleFactory> */
    use HasFactory;

    protected function casts(): array
    {
        return [
            'capacity' => 'integer',
            'status' => TransportStatus::class,
        ];
    }

    /**
     * @return BelongsTo<School, $this>
     */
    public function school(): BelongsTo
    {
        return $this->belongsTo(School::class);
    }

    /**
     * The route this vehicle currently serves, if any.
     *
     * @return HasOne<TransportRoute, $this>
     */
    public function route(): HasOne
    {
        return $this->hasOne(TransportRoute::class);
    }

    /**
     * @return HasMany<TransportTrip, $this>
     */
    public function trips(): HasMany
    {
        return $this->hasMany(TransportTrip::class);
    }
}
